test/xbgpu/test_engine.py

```diff
@@ -748,6 +905,7 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
                 frequency=frequency,
                 n_spectra_per_heap=n_spectra_per_heap,
             )
+            last_timestamp = bpipeline._weights_steady
```

Comment:
I had similar unease about doing this. Still, fair point.
* The `first_timestamp` will 'always be zero' with the test in its current guise because (as per my comment)
  * `# The first timestamp should be zero as the xbengine has not been given data to process yet`
  * I'll update that comment to be more explicit that the engine is idling.